k-distribution/include/builtin/domains.k

```diff
@@ -836,6 +836,9 @@ module BYTES-IN-K
   rule padLeftBytes(BS, LEN, VAL) => BS requires lengthBytes(BS) >=Int LEN
   rule padLeftBytes(BS, LEN, VAL) => padLeftBytes(VAL : BS, LEN, VAL) requires lengthBytes(BS) <Int LEN
```

Comment:
Judging by K implementation there are no restrictions on middle argument. Hooked implementation must be reviewed and adapted so that it matches K implementation. If this is not practical then the discrepancy must be commented.

Reply:
We should change the K implementation so that the len requirement has to be nonnegative.